modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt

```diff
@@ -156,14 +161,32 @@ class MirrordExecManager(private val service: MirrordProjectService) {
         return Pair(executionInfo.environment, executionInfo.patchedPath)
     }
 
-    class Wrapper(private val manager: MirrordExecManager, private val product: String) {
+    /**
+     * Wrapper around `MirrordExecManager` that is called by each IDE, or language variant.
+     *
+     * Helps to handle special cases and differences between the IDEs or language runners (like npm).
+     */
+    class Wrapper(private val manager: MirrordExecManager, private val product: String, private val extraEnvVars: Map<String, String>?) {
+
+        /**
+         * These are the env vars set in the launch project config, and those that are set in
+         * some other ways, such as a patched command line, or as arguments to some app.
+         *
+         * Initialized only once to avoid issues when the user could change the active mirrord
+         * config mid-run, as we mix `MIRRORD_`-style env vars with the values in the `mirrord.json`
+         * file.
+         */
+        private val projectEnvVars: Map<String, String>? = run {
```

Comment:
Wait, aren't `extraEnvVars` supposed to be taken from the run config? For example, in `GolandRunConfigurationExtension` you pass `configuration.getCustomEnvironment()` as `extraEnvVars`. Imho you don't need to read env from active launch settings, everything you need is already contained in the run config.

Reply:
So the `params.env` and `ExternalSystemRunConfiguration` contain these launch env vars?

Reply:
Also, aren't `extraEnvVars` coming only from the Idea launcher? The other IDEs don't have the same `params`.